src/agent/block_store_services/block_store_fs.js

```diff
@@ -230,22 +254,39 @@ class BlockStoreFs extends BlockStoreBase {
                 usage.count -= 1;
 
                 // also make sure we do not leave old .meta files on overwrite
-                const meta_path = this._get_block_meta_path(block_md.id);
                 const overwrite_meta_stat = await nb_native().fs.stat(fs_context, meta_path).catch(ignore_not_found);
                 if (overwrite_meta_stat) {
                     usage.size -= overwrite_meta_stat.size;
-                    await nb_native().fs.unlink(fs_context, meta_path).catch(ignore_not_found);
+                    if (!this.xattr_fallback) {
+                        // Remove the file iff we are not in
+                        await nb_native().fs.unlink(fs_context, meta_path).catch(ignore_not_found);
+                    }
                 }
             }
         }
 
         try {
-            await nb_native().fs.writeFile(fs_context, block_path, data, {
-                xattr,
-                xattr_try,
-                xattr_need_fsync,
-            });
+            if (this.xattr_fallback) {
+                await Promise.all([
+                    nb_native().fs.writeFile(fs_context, block_path, data),
+                    nb_native().fs.writeFile(fs_context, meta_path, Buffer.from(block_md_data, 'utf8')),
+                ]);
+            } else {
+                await nb_native().fs.writeFile(fs_context, block_path, data, {
+                    xattr: { [config.BLOCK_STORE_FS_XATTR_BLOCK_MD]: block_md_data },
+                    xattr_try,
+                    xattr_need_fsync,
+                });
+            }
         } catch (err) {
+            if (err.code === 'ENOTSUP') {
```

Comment:
again, we should check that xattr_fallback is not already set so that we don't get into an infinite loop in case of any other kind of ENOTSUP case -

```suggestion
            if (err.code === 'ENOTSUP' && !this.xattr_fallback) {
```